runtime/src/tasks/fifo/tasks-fifo.c

```diff
@@ -310,6 +312,8 @@ void chpl_task_init(void) {
   setup_main_thread_private_data();
 
   initialized = true;
+
+  warn_threads_per_locale();
```

Comment:
I would move the warning into `chpl_thread_init` where we call `chpl_task_getenvNumThreadsPerLocale`. Maybe something like:

```c
if ((lim = chpl_task_getenvNumThreadsPerLocale()) > 0) {
  chpl_task_warnNumThreadsPerLocale("Setting number of threads in CHPL_TASKS=fifo can lead to deadlock")
  maxThreads = lim;
}
```

And then` chpl_task_warnNumThreadsPerLocale` would exist as a sibling to `chpl_task_getenvNumThreadsPerLocale`. We could use this warning wrapper in `chpl_task_getenvNumThreadsPerLocale` itself if threads exceeds comm limit and in qthreads  `setupAvailableParallelism` where we currently warn (though for some reason not using `chpl_warning`) if `chpl_task_getenvNumThreadsPerLocale` is higher than the number of logical cores.

Reply:
If I move it into `chpl_thread_init`, what mechanism can I use to tell if CHPL_TASKS=fifo? I looked for a macro etc but didn't see anything

Reply:
`CHPL_TASKS=fifo` requires `CHPL_THREADS=pthreads` and vice-versa. It used to be that we had different threading layers under fifo, but that's not true anymore and ideally we'd just merge them into a single layer, but it hasn't been worth the time to make that change.

Reply:
And technically you could check if `CHPL_TASKS_FIFO` is defined (set by `chpl-env-gen.h`), but I don't think there's any reason to.

Reply:
Updated with that suggestion. Didn't touch qthreads though